driver-core/src/main/com/mongodb/internal/connection/netty/NettyStream.java

```diff
@@ -378,7 +381,7 @@ public void close() {
             for (Iterator<io.netty.buffer.ByteBuf> iterator = pendingInboundBuffers.iterator(); iterator.hasNext();) {
                 io.netty.buffer.ByteBuf nextByteBuf = iterator.next();
                 iterator.remove();
-                nextByteBuf.release();
+                nextByteBuf.release(nextByteBuf.refCnt());
```

Comment:
As i see, we retain a buffer at most three times in `NettyStream`:

-  Initial ByteBuf from the allocator
-  When adding to pendingInboundBuffers
-  When adding retainedSlice to CompositeBuffer, that is returned from read/readAsync.

The first two are internal, but the third is owned by the caller. As long as we keep the invariant we won't hit `IllegalReferenceCountException`:
```
Every ByteBuf returned by NettyStream.read*() **must** be
release()‑d before NettyStream.close() is invoked.
```
As i see, all call‑sites satisfy the invariant today, and I don’t see it changing. I suggest adding an explicit comment to document this for the future:
```suggestion
                // Drops all retains to prevent silent leaks; assumes callers have already released   
                // ByteBuffers returned by that NettyStream before calling close.
                nextByteBuf.release(nextByteBuf.refCnt());
```